src/components/EReceiptThumbnail.js

```diff
@@ -0,0 +1,124 @@
+import React, {useState} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import ONYXKEYS from '../ONYXKEYS';
+import * as StyleUtils from '../styles/StyleUtils';
+import transactionPropTypes from './transactionPropTypes';
+import styles from '../styles/styles';
+import * as Expensicons from './Icon/Expensicons';
+import * as MCCIcons from './Icon/MCCIcons';
+import Icon from './Icon';
+import * as ReportUtils from '../libs/ReportUtils';
+import variables from '../styles/variables';
+import * as eReceiptBGs from './Icon/EReceiptBGs';
+import Image from './Image';
+import CONST from '../CONST';
+
+const propTypes = {
+    /* TransactionID of the transaction this EReceipt corresponds to */
+    // eslint-disable-next-line react/no-unused-prop-types
+    transactionID: PropTypes.string.isRequired,
+
+    /* Onyx Props */
+    transaction: transactionPropTypes,
+};
+
+const defaultProps = {
+    transaction: {},
+};
+
+const backgroundImages = {
+    [CONST.ERECEIPT_COLORS.YELLOW]: eReceiptBGs.EReceiptBG_Yellow,
+    [CONST.ERECEIPT_COLORS.ICE]: eReceiptBGs.EReceiptBG_Ice,
+    [CONST.ERECEIPT_COLORS.BLUE]: eReceiptBGs.EReceiptBG_Blue,
+    [CONST.ERECEIPT_COLORS.GREEN]: eReceiptBGs.EReceiptBG_Green,
+    [CONST.ERECEIPT_COLORS.TANGERINE]: eReceiptBGs.EReceiptBG_Tangerine,
+    [CONST.ERECEIPT_COLORS.PINK]: eReceiptBGs.EReceiptBG_Pink,
+};
+
+function getBackgroundImage(transaction) {
+    return backgroundImages[StyleUtils.getEReceiptColorCode(transaction)];
+}
+
+function EReceiptThumbnail({transaction}) {
```

Comment:
Yeah but dont we have to add it here same as you have it in the eReceipt.js?

Reply:
I felt like the transactionID should be here too, but its not used in the component and for withOnyx its enough when mentioned in the propTypes. so no problem